src/Gaffer/NameValuePlug.cpp

```diff
@@ -0,0 +1,157 @@
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/NameValuePlug.h"
+
+#include "Gaffer/PlugAlgo.h"
+#include "Gaffer/StringPlug.h"
+
+using namespace Imath;
+using namespace IECore;
+using namespace Gaffer;
+
+IE_CORE_DEFINERUNTIMETYPED( NameValuePlug );
+
+NameValuePlug::NameValuePlug( const std::string &name, Direction direction, unsigned flags )
+	:	ValuePlug( name, direction, flags )
+{
+}
+
+NameValuePlug::NameValuePlug( const std::string &nameDefault, const IECore::Data *valueDefault, const std::string &name, Direction direction, unsigned flags )
+	:	NameValuePlug( nameDefault, PlugAlgo::createPlugFromData( "value", direction, flags, valueDefault ).get(), name )
+{
+}
+
+NameValuePlug::NameValuePlug( const std::string &nameDefault, Gaffer::PlugPtr valuePlug, const std::string &name )
+	:	NameValuePlug( name, valuePlug->direction(), valuePlug->getFlags() )
+{
+	addChild( new StringPlug( "name", valuePlug->direction(), nameDefault, valuePlug->getFlags() ) );
+
+	valuePlug->setName( "value" );
+	addChild( valuePlug );
+}
+
+NameValuePlug::NameValuePlug( const std::string &nameDefault, const IECore::Data *valueDefault, bool enabled, const std::string &name, Direction direction, unsigned flags )
+	:	NameValuePlug( nameDefault, valueDefault, name, direction, flags )
+{
+	addChild( new BoolPlug( "enabled", direction, enabled ) );
+}
+
+NameValuePlug::NameValuePlug( const std::string &nameDefault, Gaffer::PlugPtr valuePlug, bool enabled, const std::string &name )
+	:	NameValuePlug( nameDefault, valuePlug, name )
+{
+	addChild( new BoolPlug( "enabled", valuePlug->direction(), enabled, valuePlug->getFlags() ) );
+}
+
+// We need to check if the namePlug exists because we offer a bare constructor that leaves the child plugs
+// uninitialized, for backwards compatibility with the deprecated CompoundDataPlug.MemberPlug
+StringPlug *NameValuePlug::namePlug()
+{
+	return children().size() > 0 ? getChild<StringPlug>( 0 ) : nullptr;
+}
+
+const StringPlug *NameValuePlug::namePlug() const
+{
+	return children().size() > 0 ? getChild<StringPlug>( 0 ) : nullptr;
+}
+
+BoolPlug *NameValuePlug::enabledPlug()
+{
+	return children().size() > 2 ? getChild<BoolPlug>( 2 ) : nullptr;
+}
+
+const BoolPlug *NameValuePlug::enabledPlug() const
+{
+	return children().size() > 2 ? getChild<BoolPlug>( 2 ) : nullptr;
+}
+
+bool NameValuePlug::acceptsChild( const Gaffer::GraphComponent *potentialChild ) const
+{
+	if( !Plug::acceptsChild( potentialChild ) )
```

Comment:
Ah, since we're deriving from `ValuePlug`, we should be calling `ValuePlug::acceptsChild()` here, not `Plug::acceptsChild()`. I guess this is why you're getting away with accepting `PlugPtr value` in the constructor. So I think it's looking like we need to derive from Plug?